src/elements/types.rs

```diff
@@ -146,6 +146,9 @@ impl Serialize for BlockType {
 pub struct FunctionType {
 	form: u8,
 	params: Vec<ValueType>,
+	#[cfg(feature="multi_value")]
+	results: Vec<ValueType>,
```

Comment:
you can leave `Vec<ValueType>` here
just don't allow to be it more than of length 1 based on feature activated

Reply:
Than you won't need all other conditional compilations flags, just one condition at the deserialisation and serialization

Reply:
Do we want to get rid of `FunctionType::return_type_mut` and `FunctionType::return_type` then too? It wouldn't be possible to have the former if there is no longer an Option to reference within the FunctionType struct.

Reply:
Yep